server/src/http/broadcaster.rs

```diff
@@ -101,88 +124,88 @@ impl Broadcaster {
     async fn heartbeat(&self) {
         let mut active_connections = 0i64;
         for mut group in self.active_connections.iter_mut() {
-            let mut ok_clients = Vec::new();
+            let clients = std::mem::take(&mut group.clients);
+            let ok_clients = &mut group.clients;
 
-            for client in &group.clients {
-                if client
+            for ClientData { token, sender } in clients {
+                if sender
                     .send(sse::Event::Comment("keep-alive".into()))
                     .await
                     .is_ok()
                 {
-                    ok_clients.push(client.clone());
+                    ok_clients.push(ClientData { token, sender });
```

Comment:
These are memory shenanigans that we probably don't need, but might be nice?

I initially wanted to just avoid cloning each client (especially now that we also have a string in there). To do that, we'd have to take ownership, which [`std::mem::take`](https://doc.rust-lang.org/std/mem/fn.take.html) lets you do. 

This way, we take ownership of the clients and swap it in with an empty vec. We then mutably borrow that new vec and insert only the ok clients. 

Love to have a discussion about this, but it appears to be working.

Reply:
I don't think memory shenanigans are worth it here. I'm late to this party but unless this runs every time a connected client tries to get features or the client struct holds a non Arc'd struct that's a direct chunk of memory for all the features then simpler is better.

Profile first either way

Reply:
Yeah, that's fair. This runs every 30 seconds for every client. Again, probably not gonna make a massive difference, so I'd be happy to revert it to how it was 🤷🏼  I don't necessarily want to go through the rigmarole of profiling for this right away, though I'd be happy to have a look at it with you, if you want to have a look at the heaptrack output. I ... don't know what that is, so an intro might be in order 😅

Reply:
Yuh, we can take a look together, Heaptrack is this beast: https://github.com/KDE/heaptrack, give you insight into the heap allocations. It's not urgent and a quick and dirty "glance at the process memory in task manager is probably good enough for now but we should know how much memory this is going to eat before we release it

Reply:
Ah, neat. Yeah, if that's a Linux-only tool, then I'm a little out of luck. Looks like it may be possible to compile it for macOS, but yeah, let's not start there. I'd love to have a little sesh on it if you're down, though. Just hit me up whenever.